### PR TITLE
Create new licence document

### PR DESCRIPTION
**Pull request by 1pretz1**
This PR is concerned with setting up the basic fields / schema / templates needed to create a licence in Specialist Publisher, there will be follow up tasks, e.g adding the ability for these documents to be rendered in Frontend / adding model validations etc. 

We need to setup the new document type as we're looking to migrate away from the current [Licence Finder](https://www.gov.uk/licence-finder). We'll be running user testing against the new licence finder created by Specialist Publisher.

The test build will fail until https://github.com/alphagov/publishing-api/pull/2211 is merged. 

## Screenshots

Screenshot 2022-12-13 at 23 34 30

Screenshot 2022-12-13 at 23 35 02

Screenshot 2022-12-13 at 19 25 38

Screenshot 2022-12-13 at 19 28 13

Trello:
https://trello.com/c/mGJ5DIsI/1659-create-licence-document-type-in-specialist-publisher

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: spec/features/licence_transaction_spec.rb
Comment (baisa) [inline spec/features/licence_transaction_spec.rb:52]: Maybe consider passing payload value here?

**Review by baisa — COMMENTED** [review]
Looks good @1pretz1 ! Happy to approve when this is ready to merge 👍  (We just need to remember to set primary_publishing_organisation and remove pre production flag).

File: spec/features/licence_transaction_spec.rb
Comment (1pretz1) [reply spec/features/licence_transaction_spec.rb:52]: Nice suggestion, I did ponder about testing the payload (like some other features) but decided against it for a couple of reasons:

1) It is a bit strange to be testing API calls / payload in a feature test, as the expectations should just be what appears on the page versus what the underlying code is doing (api calls, model updates etc). Feels like asserting a status and a generic call is kind of okay, but asserting a full payload seems too much.

2) The payload / data structure is already tested in other places, e.g the [model is validated against Publishing API's schema]( https://github.com/alphagov/specialist-publisher/pull/2169/files) and the [shared payload presenter has a spec]( https://github.com/alphagov/specialist-publisher/blob/main/spec/presenters/document_presenter_spec.rb).

I've also tested various scenarios against the feature test, like removing a model attribute, updating a field name etc which all results in the feature test failing so pretty confident we don't need to assert the payload too.

What are your thoughts?

**Comment by 1pretz1** [comment]
Thanks for the review @baisa! I believe I've addressed your comment, feel free to disagree though! 

True, we'll need to update the `primary_publishing_organisation` and remove pre production flag if / when the finder goes officially live. However, for User Research, we'll be keeping the primary org as GDS and setting the pre-production flag to ensure the finder doesn't get published after we merge the PR.

File: lib/documents/schemas/licence_transactions.json
Comment (baisa) [inline lib/documents/schemas/licence_transactions.json:13]: Just a note regarding how results order is determined - we can customise it by adding for example `"default_order": "title",` , see more examples here https://github.com/alphagov/specialist-publisher/search?q=default_order

File: spec/features/licence_transaction_spec.rb
Comment (baisa) [reply spec/features/licence_transaction_spec.rb:52]: Sounds reasonable 👍

**Review by baisa — APPROVED** [review]
Ok sure @1pretz1 , we can merge with pre production flag if we want this deployed for User Research 👍

File: lib/documents/schemas/licence_transactions.json
Comment (1pretz1) [reply lib/documents/schemas/licence_transactions.json:13]: Ah interesting, didn't know that, thanks! I can't see an obvious ordering to apply so I'll leave this for now but will keep this in mind if there is a user need.

**Pull request merged**
